sdk/include/opentelemetry/sdk/trace/sampler.h

```diff
@@ -0,0 +1,91 @@
+#pragma once
+
+#include "opentelemetry/common/attribute_value.h"
+#include "opentelemetry/nostd/span.h"
+#include "opentelemetry/nostd/string_view.h"
+#include "opentelemetry/sdk/common/atomic_shared_ptr.h"
+#include "opentelemetry/trace/span.h"
+#include "opentelemetry/trace/trace_id.h"
+#include "opentelemetry/trace/tracer.h"
+#include "opentelemetry/version.h"
+
+#include <map>
+#include <memory>
+#include <string>
+
+OPENTELEMETRY_BEGIN_NAMESPACE
+namespace sdk
+{
+namespace trace
+{
+namespace trace_api = opentelemetry::trace;
+/**
+ * NOT_RECORD - IsRecording() == false,
+ *    span will not be recorded and all events and attributes will be dropped.
+ * RECORD - IsRecording() == true, but Sampled flag MUST NOT be set.
+ * RECORD_AND_SAMPLED - IsRecording() == true AND Sampled flag` MUST be set.
+ */
+enum class Decision
+{
+  NOT_RECORD,
+  RECORD,
+  RECORD_AND_SAMPLE
+};
+
+/**
+ * A placeholder for common::AttributeKeyValue
+ * A key/value pair that can be used to set attributes.
+ */
+struct AttributeKeyValue {
+    nostd::string_view key;
+    common::AttributeValue value;
+};
+
+struct SamplingResult
+{
+  Decision decision;
+  // TODO: Change AttributeKeyValue to common::AttributeKeyValue
+  std::unique_ptr<nostd::span<AttributeKeyValue>> attributes;
```

Comment:
why not just std::vector?

Reply:
Because the `Span` uses the same data structure. https://github.com/open-telemetry/opentelemetry-cpp/pull/117/files#diff-ee947b76b7c7a798b53dd096a45166a9R44